packages/mdc-theme/_functions.scss

```diff
@@ -51,3 +56,66 @@
     @return "light";
   }
 }
+
+// Return the given number if $value >= $min and $value <= $max. If $value falls
+// outside those bounds, return the nearest boundary value ($max or $min).
+@function mdc-theme-clamp-number_($value, $min, $max) {
+  @return max($min, min($max, $value));
+}
+
+// lighten() and darken() require values to be between 0% and 100%.
+@function mdc-theme-clamp-lighten-amount_($percentage) {
```

Comment:
I split this into two functions to make it more readable, but I think you're right that they can probably just be combined.